nexusutils/dataconverter/readers/em_spctrscpy/utils/em_nexus_plots.py

```diff
@@ -0,0 +1,195 @@
+    """Choose a preferred NXdata/data instance for ADF."""
+    entry_name = "entry" + str(entry_id)
+    trg = "/ENTRY[" + entry_name + "]/measurement/EVENT_DATA_EM[event_data_em1]/"
+    trg += "IMAGE_SET_EM_ADF[image_set_em_adf1]/"
+
+    path = ""
+    if trg + "stack/DATA[data_counts]" in template.keys():
+        assert isinstance(
+            template[trg + "stack/DATA[data_counts]"]["compress"], np.ndarray), \
+            "ADF data stack not existent!"
+        path = "stack"
+
+    if path != "":
+        print("Found adf default plot for entry " + entry_name + " at " + path)
+        trg = "/"
+        template[trg + "@default"] = entry_name
+        trg += "ENTRY[" + entry_name + "]/"
+        template[trg + "@default"] = "measurement"
+        trg += "measurement/"
+        template[trg + "@default"] = "event_data_em1"
+        trg += "EVENT_DATA_EM[event_data_em1]/"
+        template[trg + "@default"] = "image_set_em_adf1"
+        trg += "IMAGE_SET_EM_ADF[image_set_em_adf1]/"
+        template[trg + "@default"] = path
+        return True
+
+    return False
+
+
+def image_plot_available(template: dict, entry_id: int) -> bool:
+    """Choose a preferred NXdata/data instance for generic image."""
+    entry_name = "entry" + str(entry_id)
+    trg = "/ENTRY[" + entry_name + "]/measurement/EVENT_DATA_EM[event_data_em1]/"
+    trg += "IMAGE_SET_EM[image_set_em1]/"
+
+    path = ""
+    if trg + "stack/DATA[data_counts]" in template.keys():
+        assert isinstance(
+            template[trg + "stack/DATA[data_counts]"]["compress"], np.ndarray), \
+            "Generic image data stack not existent!"
+        path = "stack"
+
+    if path != "":
+        print("Found image default plot for entry " + entry_name + " at " + path)
+        trg = "/"
+        template[trg + "@default"] = entry_name
+        trg += "ENTRY[" + entry_name + "]/"
+        template[trg + "@default"] = "measurement"
+        trg += "measurement/"
+        template[trg + "@default"] = "event_data_em1"
+        trg += "EVENT_DATA_EM[event_data_em1]/"
+        template[trg + "@default"] = "image_set_em1"
+        trg += "IMAGE_SET_EM[image_set_em1]/"
+        template[trg + "@default"] = path
+        return True
+
+    return False
+
+
+def em_spctrscpy_default_plot_generator(template: dict, n_entries: int) -> dict:  # ignore:R0915
+    """For a valid NXS file at least one default plot is required."""
+
+    for entry_id in np.arange(1, n_entries + 1):
+        # by convention showing always the first entry of an NXevent_data_em
+
+        # add ebsd
+
+        if xray_plot_available(template, entry_id) is True:
+            continue
+
+        if eels_plot_available(template, entry_id) is True:
+            continue
+
+        if adf_plot_available(template, entry_id) is True:
+            continue
+
+        # add bse, se
+
+        if image_plot_available(template, entry_id) is True:
+            continue
```

Comment:
The logic could be reduced to a list of returned booleans and checked for any to be not True. That's much shorter.

Reply:
No this will not be the same, these are executed inside a loop of NXentries each with eventually different data nested under measurements, if e.g. the first NXentry has an xray in it plot is linked we're going to the next NXentry, so there is no need to check necessarily for each entry if all the plots are there because we can anyway make only one the default shown upon starting h5web.